src/main/java/com/redhat/engineering/srcclr/processor/CVEProcessor.java

```diff
@@ -48,15 +49,17 @@
 
             if ( isNotEmpty ( vuln.getCve() ) )
             {
-                SecurityDataResult securityDataResult = securityDataProcessor.process( vuln.getCve() );
-                securityDataResult.updateMessage( vuln );
-
-                matched.put( vuln, securityDataResult.getNotify());
+                ProcessorResult processorResult = securityDataProcessor.process( vuln.getCve() );
+                processorResult.setVulnerability (vuln);
+                processorResult.setLibrary (library);
+                processorResult.setScanReport( record.getMetadata() );
+                processorResult.updateMessage( vuln );
```

Comment:
now that ProcessorResult have Vulnerability inside, updateMessage() doesn't need to be called separately. It could be done when setting the vuln.